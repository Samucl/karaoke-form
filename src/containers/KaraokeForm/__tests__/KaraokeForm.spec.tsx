import '@testing-library/jest-dom'
import { render, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { KaraokeForm } from '..'

describe('KaraokeForm Component', () => {
    beforeEach(() => {
        render(<KaraokeForm />)
    })

    it('renders correctly', () => {
        expect(screen.getByText('Ilmoittautumislomake')).toBeInTheDocument()
        expect(screen.getByText('Nimi tai nimimerkki*')).toBeInTheDocument()
        expect(screen.getByText('Kasvokuva')).toBeInTheDocument()
        expect(screen.getByText('Biisi*')).toBeInTheDocument()
        expect(screen.getByText('S채vellaji*')).toBeInTheDocument()
        expect(
            screen.getByText(
                'Sallin tietojeni tallennuksen karaokej채rjestelm채채n'
            )
        ).toBeInTheDocument()
        expect(
            screen.getByRole('button', { name: 'Ilmoittaudu' })
        ).toBeEnabled()
    })

    it('shows an error message if the submit button is clicked with empty required fields', async () => {
        userEvent.click(screen.getByRole('button', { name: 'Ilmoittaudu' }))
        await waitFor(() => {
            expect(screen.getByText('Anna nimi')).toBeInTheDocument()
            expect(
                screen.getByRole('button', { name: 'Ilmoittaudu' })
            ).toBeEnabled()
        })
    })

    it('matches the snapshot', () => {
        const { asFragment } = render(<KaraokeForm />)
        expect(asFragment()).toMatchSnapshot()
    })
})
