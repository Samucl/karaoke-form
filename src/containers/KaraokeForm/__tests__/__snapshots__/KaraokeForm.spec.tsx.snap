// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KaraokeForm Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-khjJXk dHyDhO"
  >
    <h4>
      Ilmoittautumislomake
    </h4>
    <form
      class="sc-dJGMql cvqFQP"
    >
      <div
        class="sc-blmEgr dehJPy"
      >
        <span
          class="sc-hLQSwg ZxWvM"
        >
          <strong>
            Nimi tai nimimerkki*
          </strong>
        </span>
        <input
          class="sc-ifyrAs ckspDp"
          name="name"
          type="text"
        />
      </div>
      <div
        class="sc-gLLuof hKelyI"
      >
        <span
          class="sc-hLQSwg ZxWvM"
        >
          <strong>
            Kasvokuva
          </strong>
        </span>
        <div
          class="sc-iBdnpw iqTFZJ"
        >
          <input
            accept="image/*"
            class="sc-fsYfdN bPgnkf"
            id="file"
            name="file"
            type="file"
          />
          <label
            class="sc-csKJxZ jKqWbh"
            for="file"
          >
            + Tuo kasvokuva
          </label>
        </div>
      </div>
      <div
        class="sc-eDLKkx rrIuQ"
      >
        <span
          class="sc-hLQSwg ZxWvM"
        >
          <strong>
            Biisi*
          </strong>
        </span>
        <select
          class="sc-jTQCzO jjzoCu"
          name="song"
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Valitse alta
          </option>
          <option
            value="0"
          >
            Whitney Houston - I Will Always Love You
          </option>
          <option
            value="1"
          >
            Journey - Don't Stop Believin'
          </option>
          <option
            value="2"
          >
            Queen - Bohemian Rhapsody
          </option>
          <option
            value="3"
          >
            Bon Jovi - Livin' on a Prayer
          </option>
          <option
            value="4"
          >
            ABBA - Dancing Queen
          </option>
        </select>
      </div>
      <div
        class="sc-brSamD cbnvnW"
      >
        <span
          class="sc-hLQSwg ZxWvM"
        >
          <strong>
            S채vellaji*
          </strong>
        </span>
        <div
          class="sc-ktwOfi kkrBVD"
        >
          <label
            class="sc-fmKFGs gFOmcv"
          >
            <input
              class="sc-hABBmJ iNCQxZ"
              name="key"
              type="radio"
              value="-2"
            />
            <span
              class="sc-fHejqy bsvagH"
            >
              -2
            </span>
          </label>
          <label
            class="sc-fmKFGs gFOmcv"
          >
            <input
              class="sc-hABBmJ iNCQxZ"
              name="key"
              type="radio"
              value="-1"
            />
            <span
              class="sc-fHejqy bsvagH"
            >
              -1
            </span>
          </label>
          <label
            class="sc-fmKFGs gFOmcv"
          >
            <input
              class="sc-hABBmJ iNCQxZ"
              name="key"
              type="radio"
              value="0"
            />
            <span
              class="sc-fHejqy bsvagH"
            >
              0
            </span>
          </label>
          <label
            class="sc-fmKFGs gFOmcv"
          >
            <input
              class="sc-hABBmJ iNCQxZ"
              name="key"
              type="radio"
              value="1"
            />
            <span
              class="sc-fHejqy bsvagH"
            >
              1
            </span>
          </label>
          <label
            class="sc-fmKFGs gFOmcv"
          >
            <input
              class="sc-hABBmJ iNCQxZ"
              name="key"
              type="radio"
              value="2"
            />
            <span
              class="sc-fHejqy bsvagH"
            >
              2
            </span>
          </label>
        </div>
      </div>
      <div
        class="sc-beySPh jZXJSS"
      >
        <input
          class="sc-guDLey beTZKc"
          name="box"
          type="checkbox"
        />
        <span
          class="sc-dmyCSP bYODjd"
        >
          Sallin tietojeni tallennuksen karaokej채rjestelm채채n
        </span>
      </div>
      <button
        class="sc-hIPBNq dkghb"
        type="submit"
      >
        Ilmoittaudu
      </button>
    </form>
  </div>
</DocumentFragment>
`;
